crates/fuel-core/src/service.rs

```diff
@@ -196,9 +196,10 @@ pub struct Task {
 
 impl Task {
     /// Private inner method for initializing the fuel service task
-    pub fn new(database: Database, config: Config) -> anyhow::Result<Task> {
+    pub fn new(database: Database, mut config: Config) -> anyhow::Result<Task> {
         // initialize state
         genesis::maybe_initialize_state(&config, &database)?;
+        config.chain_conf.initial_state = None;
```

Comment:
What's the `initial_state` before this line? Isn't there any other way for this to get initialized to desired state or it has to go from Some<T> to None in here?